hypershift-operator/metrics.go

```diff
@@ -117,6 +139,38 @@ func newMetrics(client crclient.Client, log logr.Logger) *hypershiftMetrics {
 			Name: "hypershift_nodepools_size",
 			Help: "Number of replicas associated with a given NodePool",
 		}, []string{"name", "platform"}),
+		// hypershiftOperatorInfo is a metric to capture the current operator details of the management cluster
+		hypershiftOperatorInfo: prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+			Name: "hypershift_operator_info",
+			Help: "Number in seconds for the Pod lastTransitionTime to Ready state. Also detailed info about the operator.",
+			ConstLabels: prometheus.Labels{
+				"version":                hypershiftVersion,
+				"image":                  image,
+				"imageId":                imageId,
+				"latestSupportedVersion": latestSupportedVersion,
+				"goVersion":              goVersion,
+				"goArch":                 goArch,
+			},
+		}, func() float64 {
+			ctx := context.TODO()
+			hypershiftNamespace := os.Getenv("MY_NAMESPACE")
+			hypershiftPodName := os.Getenv("MY_NAME")
+			var lastTransitionTime time.Time
+
+			hypershiftPod := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: hypershiftPodName, Namespace: hypershiftNamespace}}
+			if err := client.Get(ctx, crclient.ObjectKeyFromObject(hypershiftPod), hypershiftPod); err != nil {
+				return float64(0)
+			}
+
+			// Get the last transition time
+			for _, c := range hypershiftPod.Status.Conditions {
+				if c.Type == corev1.PodReady {
+					lastTransitionTime = c.LastTransitionTime.Time
+				}
+			}
+
+			return float64(lastTransitionTime.Second())
```

Comment:
why do we need this, why not just return a 1? The value of this metric is irrelevant.

Reply:
As we discussed here, looks like could be important: https://github.com/openshift/hypershift/pull/2443#discussion_r1170467806

Reply:
@csrwng @enxebre, WDYT?

Reply:
yeh there's little/no value for lastTransitionTime, feel free to change it to 1 and we can go from there.

Reply:
It could make sense to measure how much time our operator takes to be ready. This measure could make sense if you have a history about a progression. Thinking mostly in how the app has been evolving with the new code and  how the booting time was increased/decreased.

I think I will maintain it for know as it I have here, and if does not make sense for the future, I can submit a new PR just returning 1.